datasource/opennms-direct/src/main/java/org/opennms/alec/datasource/opennms/jvm/ApiMapper.java

```diff
@@ -185,41 +185,48 @@ public InMemoryEvent toEvent(Situation situation) {
         final ImmutableInMemoryEvent.Builder eventBuilder = ImmutableInMemoryEvent.newBuilder();
         eventBuilder.setUei(SITUATION_UEI).setSource("alec");
 
-        // Use the max severity as the situation severity
-        final Severity maxSeverity = Severity.fromValue(situation.getAlarms().stream()
-                .mapToInt(a -> a.getSeverity() != null ? a.getSeverity().getValue() : Severity.INDETERMINATE.getValue())
-                .max()
-                .orElseGet(Severity.INDETERMINATE::getValue));
-        eventBuilder.setSeverity(toSeverity(maxSeverity));
+        if(situation.getAlarms().isEmpty() && situation.getStatus() == Status.REJECTED) {
+            //Situation should be cleared
+            eventBuilder.setSeverity(org.opennms.integration.api.v1.model.Severity.CLEARED);
 
-        // Relay the situation id
-        eventBuilder.addParameter(ImmutableEventParameter.newInstance(SITUATION_ID_PARM_NAME, situation.getId()));
-
-        // Populate logmsg, descr and related node with details from the most relevant alarm
-        final Alarm alarmForDescr = AlarmUtil.getAlarmForDescription(situation.getAlarms());
-        if (alarmForDescr != null) {
-            eventBuilder.addParameter(ImmutableEventParameter.newInstance("situationLogMsg",
-                    alarmForDescr.getSummary()));
-
-            String description = alarmForDescr.getDescription();
-            if (situation.getDiagnosticText() != null) {
-                description += "\n<p>ALEC Diagnostic: " + situation.getDiagnosticText() + "</p>";
+            eventBuilder.addParameter(ImmutableEventParameter.newInstance("situationLogMsg", "situation rejected"));
```

Comment:
I wonder if this shouldn't be constants?

Reply:
yep, done.